README.md

```diff
@@ -0,0 +1,194 @@
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/redux-mock-store-await-actions.svg
+[greenkeeper-image]:https://badges.greenkeeper.io/moxystudio/redux-mock-store-await-actions.svg
+[greenkeeper-url]:https://greenkeeper.io
+[conventional-commits-url]:https://img.shields.io/badge/Conventional%20Commits-1.0.0-yellow.svg
+
+> Waits for actions dispatched asynchronously until a timeout expires.
+
+**NOTE:** This module only works with [redux-mock-store](https://github.com/arnaudbenard/redux-mock-store) and shall only be used for testing purposes. Support for real Redux store is not provided.
+
+## Installation
+
+`$ npm install redux-mock-store-await-actions --save-dev`
+
+## Motivation
+
+Consider the following example:
+
+```js
+function login(username, password) {
+    return async (dispatch) => {
+        dispatch({ type: 'LOGIN_START', payload: { username, password } });
+
+        try {
+            const user = await fetch('/login', {
+                headers: { 'Content-Type': 'application/json' },
+                method: 'POST',
+                body: JSON.stringify({ username, password })
+            });
+
+            dispatch({ type: 'LOGIN_SUCCESS', payload: user });
+        } catch (err) {
+            dispatch({ type: 'LOGIN_FAIL', payload: err });
+            throw err;
+        }
+
+        // Fetch orders asynchronously, not waiting for them to be retrieved
+        dispatch(fetchOrders());
+    }
+}
+
+function fetchOrders() {
+    return async (dispatch) => {
+        dispatch({ type: 'FETCH_ORDERS_START' });
+
+        try {
+            const orders = await fetch('/account/orders');
+
+            dispatch({ type: 'FETCH_ORDERS_SUCCESS', payload: orders });
+        } catch (err) {
+            dispatch({ type: 'FETCH_ORDERS_FAIL', payload: err });
+            throw err;
+        }
+    }
+}
+
+store.dispatch(login('my-username', '12345'));
+
+expect(store.getActions()).toContain([
+    'LOGIN_START',
+    'FETCH_ORDERS_SUCCESS'
+]);
+```
+
+The assertion above will fail because `FETCH_ORDERS_SUCCESS` will not yet exist in the stack of actions.
+To solve this, one can use `setTimeout` explicitly in each test:
+
+```js
+store.dispatch(login('my-username', '12345'));
+
+setTimeout(() => expect(store.getActions()).toContain([
+    'LOGIN_START',
+    'FETCH_ORDERS_SUCCESS'
+]), 50);
+```
+
+However, this is not pretty and is error-prone. `redux-mock-store-await-actions` makes this easier for you.
+
+## Usage
+
+### Example #1: action types
+
+Supply the action types to await for.
+
+```js
+import configureStore from 'redux-mock-store';
+import thunkMiddleware from 'redux-thunk';
+
+const store = configureStore([thunkMiddleware])();
+
+store.dispatch(login('my-username', '12345'));
+await waitForActions(store, ['LOGIN_START', 'FETCH_ORDERS_SUCCESS']);
+```
+
+### Example #2: action objects
+
+Supply the action objects to await for. Matches a subset of the properties of the dispatched actions.
```

Comment:
`to await for. Matches` -> `to await for, matching`

Reply:
I don't think that `Matches a subset of the properties of the dispatched actions.` is very clear. Does it mean that only some keys from the action are compared? If so, which ones? If not, what does it do exactly?

Reply:
I will add a clarification of the expected behaviour.